.changeset/rich-chairs-wink.md

```diff
@@ -2,19 +2,22 @@
 'skuba': minor
 ---
 
-lint: Replace `.buildkite/` files with duplicated YAML merge keys, for example:
+format, lint: Fix duplicated YAML merge keys in `.buildkite/` pipelines
 
-```yaml
-# Before
-- <<: *deploy
-  <<: *docker
-  label: stuff
-
-# After
-- <<: [*deploy, *docker]
-  label: stuff
+```diff
+- - <<: *deploy
+-   <<: *docker
++ - <<: [*deploy, *docker]
+    label: stuff
 ```
 
-This should have no functional change, and is to support standardised YAML parsing across different tools, including the latest ESLint upgrades.
+This change supports standardised YAML parsing across tools such as ESLint; it should not functionally alter the behaviour of your build pipeline.
+
+The bundled patch is fairly conservative and will not attempt to migrate more complex scenarios, such as where there are other keys between the merge keys. Review whether the order of keys matters before manually updating other occurrences.
 
-This migration will not be capture all cases of this (e.g. if there are keys between the merge keys). If you have other cases, update them following the example above.
+```diff
+- - <<: *deploy
++ - <<: [*deploy, *docker]
+    label: stuff
+-   <<: *docker
+```
```

Comment:
I made this up. Does order matter?

Reply:
Yes, it does; 

![image](https://github.com/user-attachments/assets/ce9ec48e-4cb7-454e-9bdc-12aeda004585)

Reply:
Ah, I was worried about these being different:

```yaml
config:
  - &a
    a: true
  - &b
    b: true

result:
  - <<: *a
    b: false
    <<: *b
```

```yaml
config:
  - &a
    a: true
  - &b
    b: true

result:
  - <<: [*a, *b]
    b: false
```

https://codebeautify.org/yaml-editor-online seems to indicate that the merge keys are hoisted so they are not different, if I can trust it.